### ISSUE TITLE
Editing a field causes a crash in updated version (not using the internal keyboard).

### ISSUE THREAD

**Issue by VK2BEA** [PhilippC/keepass2android#2262]
With version 1.09e-r3 when I try to edit a field, the program crashes. (Android 11)
I press the edit button an then press the field to edit and it crashes. 

It only crashes if I select the external keyboard (gboard). I I use the internal keyboard its OK.
I've never had an issue with gboard on previous versions.

..... Edit

Not sure it even works with the internal keyboard... It's now crashing with this setting also.

Just reverted to last version (1.09e-r1) and confirmed that it worked without problem.

Also was not able to open database on GDrive with version 1.09e-r3 after I had deleted and reinstalled the app.

**Comment by hyproman** [comment]
I spent some time trying to reproduce this today, but without any luck. I also use gboard, but with Android 13.

When the app crashes, does Android give you a pop-up dialog box with a way to capture/send a crash/bug report? If we can get one of these with a stack-trace it would be very helpful. Depending on your technical ability, there are a few ways to potentially get this info (https://www.browserstack.com/guide/how-to-get-android-app-crash-logs).

Unfortunately the K2P Debug Logging functionality doesn't appear to capture app crashes (at least in my experience). Additionally when I enabled Android's Bug Report feature, the dialog box appeared, but the "Copy Link to details" button didn't work for some reason.

**Comment by hyproman** [comment]
It's likely there's a stack-trace somewhere buried in that zip file, but I'm not exactly sure where to look. I'd also be wary of posting it all here, in case there is sensitive info.

If you are willing to try `logcat` real-time, I generally use something like this when working with this app:
` adb logcat '*:E KP2A KP2AJ KP2A_FC KP2A_FC_P'`

My guess is what you're looking for will resemble something like this:
```
E AndroidRuntime: FATAL EXCEPTION: Thread-<#>
E AndroidRuntime: Process: keepass2android.keepass2android, PID: <###>

E AndroidRuntime: <Something>Exception: <Some message details>
E AndroidRuntime:        at some.package.and.Classname.someMethod(...)
E AndroidRuntime:        at ...
E AndroidRuntime:        at ...
E AndroidRuntime:        at ...
```

**Comment by hyproman** [comment]
I think @PhilippC is going to have to take a look at this one.

I can't quite unpack what the issue is here exactly. It's clearly an auto-fill logic problem, and it looks like there has been some recent modification/refactoring of that code in recent releases (specifically https://github.com/PhilippC/keepass2android/pull/2254).

**Comment by VK2BEA** [comment]
yes, if I disable "autofill" it doesn't crash ... but that's probably obvious 8-(

**Comment by AriehSchneier** [comment]
Maybe you can turn on [LogAutofillView](https://github.com/PhilippC/keepass2android/blob/db6b266a594c92208f7899200b5a21343a7a6c09/src/Kp2aAutofillParser/AutofillParser.cs#L816) and see what it logs just before the crash?

The exception you are getting happens if you try to add 2 items with the same key to a Dictionary. I can see 3 Dictionary add calls inside that function:
- [FieldsMappedToHints.Add](https://github.com/PhilippC/keepass2android/blob/db6b266a594c92208f7899200b5a21343a7a6c09/src/Kp2aAutofillParser/AutofillParser.cs#L841)
- [FieldsMappedToHints.Add](https://github.com/PhilippC/keepass2android/blob/db6b266a594c92208f7899200b5a21343a7a6c09/src/Kp2aAutofillParser/AutofillParser.cs#L876)
- [FieldsMappedToHints.Add](https://github.com/PhilippC/keepass2android/blob/db6b266a594c92208f7899200b5a21343a7a6c09/src/Kp2aAutofillParser/AutofillParser.cs#L878)

**Comment by AriehSchneier** [comment]
Looks like the file was copied instead of moved? I wonder which version the final code is using. The original version is missing the `.Except(usernameFields)` on this line: https://github.com/PhilippC/keepass2android/blob/db6b266a594c92208f7899200b5a21343a7a6c09/src/keepass2android/services/Kp2aAutofillParser/AutofillParser.cs#L862

**Comment by PhilippC** [comment]
thanks @VK2BEA for reporting and all others for helping to troubleshoot. This should be fixed, but I wasn't able to reproduce either, so if anybody running 1.09e-r3 can provide an AutofillView from the log which causes the crash, this would be awesome

**Comment by PhilippC** [comment]
this fix is contained in 1.09e-r4 (https://github.com/PhilippC/keepass2android/releases/tag/v1.09e-r4 or https://play.google.com/apps/testing/keepass2android.keepass2android)

**Comment by VK2BEA** [comment]
Yes, that fixed the problem for me.  Thanks!
